third_party/nvfuser/csrc/ir_internal_nodes.h

```diff
@@ -2125,6 +2133,46 @@ class TORCH_CUDA_CU_API Swizzle2D : public Expr {
   }
 };
 
+//! IterDomain expression to resize
+class TORCH_CUDA_CU_API Resize : public Expr {
```

Comment:
Just some random thoughts: Can `Resize` only be used for tensor definition? Can it be used for scheduling as well? For example, if I have horizontal fusion `T1 = T0, T3 = T2`, then I can resize the smaller one to the larger one?

Reply:
>  it's really a compute definition change, not just a schedule change.

Yes, that's why for consumer tensors, resize should only be used as rfactor. A producer tensor can be resized as demanded by its consumers. We resize a producer so that it has the same extent as its consumer. Once a producer domain has the same extent, we can propagate the consumer index to the producer.

Reply:
Isn't non-divisible split also like compute definition change? `split(I{2}, 5) -> I{1}, I{5}`

Reply:
That's a good point. Yes, it does change the extent. It won't matter for allocation correctness since we just overallocate. For indexing, what we are doing specifically for non-divisible splits basically means indexing is always done with IterDomains before non-divisible splits.